File: Utopian Tree/utopian-tree.js
/**
 * Complete the utopianTree function below.
 *
 * @param {number} n
 * @return {number}
 */
const utopianTree = n => {
  let height = 1;
  for (let i = 1; i <= n; i++) {
    if (i % 2) {
      height *= 2;
    } else {
      height++;
    }
  }

  return height;
};

console.log(utopianTree(0)); // 1
console.log(utopianTree(1)); // 2
console.log(utopianTree(4)); // 7

File: Utopian Tree/utopian-tree.py
def utopianTree(n):
    """
    :type n: int
    :rtype: int
    """
    height = 1
    for i in range(1, n + 1):
        if i % 2:
            height *= 2
        else:
            height += 1

    return height


print(utopianTree(0))  # 1
print(utopianTree(1))  # 2
print(utopianTree(4))  # 7